Modules/TRD/src/PulseHeightTrackMatch.cxx

```diff
@@ -107,172 +98,89 @@ void PulseHeightTrackMatch::startOfCycle()
-            int sec = det / 30;
-            int row = trklt.getPadRow();
-            // int col = trklt.getPadCol();
-
-            // obtain pad number relative to MCM center
-            int padLocal = trklt.getPositionBinSigned() * o2::trd::constants::GRANULARITYTRKLPOS;
-            // MCM number in column direction (0..7)
-            int mcmCol = (trklt.getMCM() % o2::trd::constants::NMCMROBINCOL) + o2::trd::constants::NMCMROBINCOL * (trklt.getROB() % 2);
-            int col = 6.f + mcmCol * ((float)o2::trd::constants::NCOLMCM) + padLocal;
-
-            for (int iDigit = trigger.getFirstDigit() + 1; iDigit < trigger.getFirstDigit() + trigger.getNumberOfDigits() - 1; ++iDigit)
-
-            {
-              const auto& digit = digits[iDigit];
-              const auto& digitBefore = digits[iDigit - 1];
-              const auto& digitAfter = digits[iDigit + 1];
-
-              std::tuple<unsigned int, unsigned int, unsigned int> aa, ba, ca;
-              aa = std::make_tuple(digitBefore.getDetector(), digitBefore.getPadRow(), digitBefore.getPadCol());
-              ba = std::make_tuple(digit.getDetector(), digit.getPadRow(), digit.getPadCol());
-              ca = std::make_tuple(digitAfter.getDetector(), digitAfter.getPadRow(), digitAfter.getPadCol());
-
-              auto [det1, row1, col1] = aa;
-              auto [det2, row2, col2] = ba;
-              auto [det3, row3, col3] = ca;
-
-              bool consecutive = false;
-              bool digitmatch = false;
-
-              if (det == det2 && row == row2) {
-
-                int coldif = col2 - col;
-                if (TMath::Abs(coldif) < 4)
-                  digitmatch = true;
-              }
-              if (det1 == det2 && det2 == det3 && row1 == row2 && row2 == row3 && col2 + 1 == col1 && col3 + 1 == col2)
-                consecutive = true;
-
-              if (digitmatch && consecutive) {
-
-                int phVal = 0;
-
-                for (int iTb = 0; iTb < o2::trd::constants::TIMEBINS; ++iTb) {
-                  phVal = digitBefore.getADC()[iTb] + digit.getADC()[iTb] + digitAfter.getADC()[iTb];
-                  mPulseHeightpro->Fill(iTb, phVal);
-                  mPulseHeightperchamber->Fill(iTb, det1, phVal);
-                }
+            if (digit.getPadRow() != trklt.getPadRow())
+              continue;
+            coldif = digit.getPadCol() - trklt.getPadCol();
+            if (TMath::Abs(coldif) > 1)
+              continue; // PadCol may be off by +-1 due to the slope of tracklet
+            const auto& digitBefore = digits[iDigit - 1];
+            const auto& digitAfter = digits[iDigit + 1];
+
+            bool consecutive = false;
+            if (digitBefore.getDetector() == digitAfter.getDetector() && digitBefore.getPadRow() == digit.getPadRow() && digit.getPadRow() == digitAfter.getPadRow() && digit.getPadCol() + 1 == digitBefore.getPadCol() && digitAfter.getPadCol() + 1 == digit.getPadCol())
```

Comment:
Did you check whether this flag `consecutive` at some point is true? And please don't forget the curly braces even around one liners
```
if (...) {
}
```

Reply:
yes, after applying this condition what remains are only those digits that are consecutive in space. I also added a comment on this condition and put curly braces on one-liners also.
Thanks.